lib/Dialect/AIEX/Transforms/AIEDmaToNpu.cpp

```diff
@@ -370,13 +370,14 @@ struct DmaToNpuPattern : OpConversionPattern<NpuDmaMemcpyNdOp> {
     buffer_offset = IntegerAttr::get(i32ty, offset);
 
     // enable_packet
+    if (auto packetInfo = op.getPacket()) {
+      enable_packet = IntegerAttr::get(i32ty, 1);
+      packet_type = IntegerAttr::get(i32ty, packetInfo->getPktType());
+      packet_id = IntegerAttr::get(i32ty, packetInfo->getPktId());
+    }
```

Comment:
This needs a test

Reply:
Sure. This PR contains new mlir tests to check the new format and verifier.

Re board test, I will land this PR, and then convert its SHIM->Memtile flow to packet flow, as board test for this feature. https://github.com/Xilinx/mlir-aie/pull/1669

Reply:
> Sure. This PR contains new mlir tests to check the new format and verifier.
> 

It also needs a mlir test for the `aie-dma-to-npu` pass update in e.g. `test/Conversion/DmaToNpu/dma_to_npu.mlir`

Reply:
Ah got it. New test added in commit [Add test to show aiex.npu.dma lowering with packet](https://github.com/Xilinx/mlir-aie/pull/1671/commits/61d20fb8bc135d3782eb210f3dd9b3eeba9ccbba)

Reply:
Board test added in commit [Packet header from shim dma board test](https://github.com/Xilinx/mlir-aie/pull/1671/commits/15eb5058d31b689e1f3888413732646be2a876ec)